### ISSUE TITLE
Make use of CDK Pipelines

### ISSUE THREAD

**Issue by Rabadash8820** [DerploidEntertainment/Website#36]
I don't know much here yet, but here's the [CDK Pipeline documentation](https://docs.aws.amazon.com/cdk/v2/guide/cdk_pipeline.html).

**Comment by Rabadash8820** [comment]
Weel, having messed with this system for a few days now, I'm actually leaning against it on the following grounds:

1. If we go "full Pipeline" with a stage for test and a stage for prod, that means the only way to see these changes in the test stage is to push to GitHub. This not only kills the inner dev loop (make a dirty commit, push it, wait for the Pipeline stack to synthesize and self-mutate, then wait for the actual stages to synthesize and deploy, make changes, amend commit, push, etc.), but also makes our git workflow really tedious and error prone (with all the amend commits, force pushing, and cherry picks of that CNAME change commit for the test repo).
2. There is no manual approval step between the test stage and the prod stage. One could be added, but this doesn't come out of the box with CDK Pipeline, it would require a deeper knowledge of AWS CodePipelines, which I have no desire to learn...
3. This would all be a _little bit_ easier if I tested my local changes against my personal account before pushing to GitHub, but for this small project I don't really want to work with multiple AWS accounts. And even then, the CDK Pipeline would only listen for pushes on the `main` branch, so it wouldn't update the test stage until we've _already merged_ a PR. I get the sense that better Pipeline-PR integration is coming, but I don't want to wait for it, and the fact that it's not here already makes CDK Pipeline as a whole feel rather immature.
4. Given that I don't want to mess with multiple AWS accounts for this, it actually makes more sense to test my local changes _against the test environment_ (i.e., the test repo with `derploid.link`). But then I would essentially need two different CDK apps, one to synthesize the CDK Pipeline app for prod, and one to synthesize the _actual_ stacks for deploying to test.

Long story short, CDK Pipeline just feels like a hacky wrapper around AWS CodePipelines, which is already a separate CI/CD system that I have no desire to learn. GitHub Actions can do everything CDK Pipeline can do (and probably more, with the GitHub Marketplace), with less setup and less research. Looks like #14 is back on the roadmap...
